Marlin/stepper_indirection.cpp

```diff
@@ -127,394 +119,70 @@
-      _2130_set_diss2g(A); \
-      _2130_set_intpol(A); \
-      _2130_set_mres(A); \
-      _2130_set_sync(A); \
-      _2130_set_vhighchm(A); \
-      _2130_set_vhighfs(A); \
-      _2130_set_vsense(A); \
-      _2130_set_tbl(A); \
-      _2130_set_chm(A); \
-      _2130_set_rndtf(A); \
-      _2130_set_disfdcc(A); \
-      _2130_set_fd(A); \
-      _2130_set_hend(A); \
-      _2130_set_hstrt(A); \
-      _2130_set_toff(A); \
-      _2130_set_sfilt(A); \
-      _2130_set_sgt(A); \
-      _2130_set_seimin(A); \
-      _2130_set_sedn(A); \
-      _2130_set_semax(A); \
-      _2130_set_seup(A); \
-      _2130_set_semin(A); \
-      _2130_set_DCCTRL(A); \
-      _2130_set_freewheel(A); \
-      _2130_set_pwm_symmetric(A); \
-      _2130_set_pwm_autoscale(A); \
-      _2130_set_pwm_freq(A); \
-      _2130_set_PWM_GRAD(A); \
-      _2130_set_PWM_AMPL(A); \
-      _2130_set_ENCM_CTRL(A); \
-    } while(0)
-
-  #else // !TMC2130_ADVANCED_CONFIGURATION
+  #define _TMC2130_DEFINE(ST) TMC2130Stepper stepper##ST(ST##_ENABLE_PIN, ST##_DIR_PIN, ST##_STEP_PIN, ST##_CHIP_SELECT)
 
-    #define _TMC2130_INIT(A) do{ \
-      stepper##A.init(); \
-      stepper##A.set_mres(A##_MRES); \
-      stepper##A.set_IHOLD_IRUN(A##_IHOLD, A##_IRUN, A##_IHOLDDELAY); \
-      stepper##A.set_I_scale_analog(A##_I_SCALE_ANALOG); \
-      stepper##A.set_tbl(A##_TBL); \
-      stepper##A.set_toff(A##_TOFF); \
-    } while(0)
-
-  #endif // TMC2130_ADVANCED_CONFIGURATION
-
-  // Stepper objects of TMC2310 steppers used
+  // Stepper objects of TMC2130 steppers used
   #if ENABLED(X_IS_TMC2130)
-    Trinamic_TMC2130 stepperX(X_CS_PIN);
+    _TMC2130_DEFINE(X);
   #endif
   #if ENABLED(X2_IS_TMC2130)
-    Trinamic_TMC2130 stepperX2(X2_CS_PIN);
+    _TMC2130_DEFINE(X2);
   #endif
   #if ENABLED(Y_IS_TMC2130)
-    Trinamic_TMC2130 stepperY(Y_CS_PIN);
+    _TMC2130_DEFINE(Y);
   #endif
   #if ENABLED(Y2_IS_TMC2130)
-    Trinamic_TMC2130 stepperY2(Y2_CS_PINR);
+    _TMC2130_DEFINE(Y2);
   #endif
   #if ENABLED(Z_IS_TMC2130)
-    Trinamic_TMC2130 stepperZ(Z_CS_PIN);
+    _TMC2130_DEFINE(Z);
   #endif
   #if ENABLED(Z2_IS_TMC2130)
-    Trinamic_TMC2130 stepperZ2(Z2_CS_PIN);
+    _TMC2130_DEFINE(Z2);
   #endif
   #if ENABLED(E0_IS_TMC2130)
-    Trinamic_TMC2130 stepperE0(E0_CS_PIN);
+    _TMC2130_DEFINE(E0);
   #endif
   #if ENABLED(E1_IS_TMC2130)
-    Trinamic_TMC2130 stepperE1(E1_CS_PIN);
+    _TMC2130_DEFINE(E1);
   #endif
   #if ENABLED(E2_IS_TMC2130)
-    Trinamic_TMC2130 stepperE2(E2_CS_PIN);
+    _TMC2130_DEFINE(E2);
   #endif
   #if ENABLED(E3_IS_TMC2130)
-    Trinamic_TMC2130 stepperE3(E3_CS_PIN);
+    _TMC2130_DEFINE(E3);
   #endif
 
+  // Use internal reference voltage for current calculations. This is the default.
+  // Following values from Trinamic's spreadsheet with values for a NEMA17 (42BYGHW609)
+  void tmc2130_init(TMC2130Stepper &st, uint16_t max_current, uint16_t microsteps) {
+    st.begin();
+    st.setCurrent(st.getCurrent(), R_SENSE, HOLD_MULTIPLIER);
```

Comment:
@teemautlut I see that the `max_current` argument isn't used here. Is it not needed here because it's set by `configuration_store.cpp`?